src/scanner-engine.ts

```diff
@@ -136,11 +136,16 @@ export function runScannerEngine(
 
   return new Promise<void>((resolve, reject) => {
     child.on('exit', code => {
-      if (code === 0) {
-        log(LogLevel.DEBUG, 'Scanner engine finished successfully');
-        resolve();
+      if (typeof code === 'number') {
+        if (code === 0) {
+          log(LogLevel.DEBUG, 'Scanner engine finished successfully');
+          resolve();
+        } else {
+          reject(new Error(`Scanner engine failed with code ${code}`));
+          process.exit(code);
```

Comment:
Nice 👍

More generally, I am thinking we should exit with an error code `1` as soon as any error bubbles up to the top-level `scan(..)` scan method. Currently, [we catch errors at top-level but only print the error](https://github.com/SonarSource/sonar-scanner-npm/blob/4579d45a06b95397b4d424a7ee54d817693bfd6c/src/scan.ts#L35), I do not see a reason why not to call `process.exit(1)` to ensure the process fails, WDYT?